gcloud-java-contrib/gcloud-java-nio/src/main/java/com/google/gcloud/storage/contrib/nio/CloudStorageFileSystemProvider.java

```diff
@@ -454,18 +520,51 @@ public FileStore getFileStore(Path path) {
   public boolean equals(@Nullable Object other) {
     return this == other
         || other instanceof CloudStorageFileSystemProvider
-        && Objects.equals(gcsService, ((CloudStorageFileSystemProvider) other).gcsService);
+        && Objects.equals(storage, ((CloudStorageFileSystemProvider) other).storage);
```

Comment:
`Storage` is an interface but I was expecting implementations to override equals. If they don't then that defaults to object equality, which may be the right thing to do here: just because we have two different objects implementing the `Service<StorageOptions>` interface with the same `StorageOptions` object doesn't guarantee that it's the same implementation, does it? For example perhaps there could be some mocking going on and one of them is an in-memory fake while the other could be the real deal.

I added a basic test for provider equality and it behaves as expected.

Perhaps leaving the code as-is isn't so bad? What do you both think?